test/distributed/tensor/test_dtensor_compile.py

```diff
@@ -464,6 +465,45 @@ def g(x):
         run(g, 64, 8)
         self.assertEqual(cnt.frame_count, 2)
 
+    @skip_unless_torch_gpu
+    def test_dtensor_unbacked_matmuls(self):
+        from torch.distributed.tensor import randn as d_randn
+
+        # use 2x2 mesh for testing
+        dist.destroy_process_group()
+        dist.init_process_group("fake", store=FakeStore(), rank=0, world_size=4)
+        device_mesh = init_device_mesh(self.device_type, (2, 2))
+
+        for test_index, (px, py) in enumerate(
+            [
+                # these should pass as no redistribute strategy is available
+                [[Replicate(), Replicate()], [Replicate(), Replicate()]],
+                [[Replicate(), Shard(0)], [Replicate(), Replicate()]],
+                [[Replicate(), Shard(1)], [Replicate(), Shard(0)]],
+            ]
+        ):
+            # create DTensors with unbacked outer/inner sizes
+            x_dt = d_randn(64, 64, device_mesh=device_mesh, placements=px)
+            y_dt = d_randn(64, 64, device_mesh=device_mesh, placements=py)
+            for i in range(2):
+                torch._dynamo.decorators.mark_unbacked(x_dt, i)
+                torch._dynamo.decorators.mark_unbacked(y_dt, i)
+
+            # full-graph capture
+            torch._dynamo.reset()
+            cnt = torch._dynamo.testing.CompileCounterWithBackend("eager")
+            fn = torch.compile(torch.mm, backend=cnt, fullgraph=True)
+            fn(x_dt, y_dt)
```

Comment:
shall we print results placements and tests against them to better understand the reults?

Reply:
yes please